extensions/cms/src/cmsUtils.ts

```diff
@@ -157,49 +229,103 @@ export class CmsUtils {
 					let result = await provider.addRegisteredServer(ownerUri, relativePath, registeredServerName, connection.options.registeredServerDescription, connection);
 					if (result) {
 						return Promise.resolve(result);
-					} else {
-						return Promise.reject(registeredServerName);
 					}
 				}
 			}
-		});
+		} catch (errorString) {
+			let error = new Error(errorString.message);
```

Comment:
Same here - why is this called errorString if you assume it's an error object below? And then again I ask why are you rethrowing it? Doing that loses the original context of where the error occurred - you should probably just be letting the error bubble up.